src/DotNetProjectFile.Analyzers/Analyzers/Files/OnlyUseUTF8WithoutBom.cs

```diff
@@ -0,0 +1,46 @@
+using Microsoft.CodeAnalysis.Text;
+
+namespace DotNetProjectFile.Analyzers.Files;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class OnlyUseUTF8WithoutBom() : ProjectFileAnalyzer<ProjectTextFile>(Rule.OnlyUseUTF8WithoutBom)
+{
+    protected override void Register(AnalysisContext context)
+    {
+        context.RegisterCompilationAction(c =>
+        {
+            if (ProjectFiles.Global.UpdateMsBuildProject(c) is { } msbuild)
+            {
+                foreach (var file in msbuild.Path.Directory.Files("**/*")?
+                    .Where(f => !f.ToString("/").Contains("/bin/"))
```

Comment:
Just a sanity check, but is is this always `/bin/` on all operating systems?

Reply:
See the `.ToString("/")`. This is 'simple' implementation. I'm working on a more robust one.